lib/core-kalman-filter.ts

```diff
@@ -80,19 +49,19 @@ class CoreKalmanFilter {
 	* @returns{Array.<Array.<Number>>}
 	*/
 
-	getPredictedCovariance(options = {}) {
+	getPredictedCovariance(options: {previousCorrected?: State, index?: number} = {}) {
 		let {previousCorrected, index} = options;
-		previousCorrected = previousCorrected || this.getInitState();
+		previousCorrected ||= this.getInitState();
 
-		const getValueOptions = Object.assign({}, {previousCorrected, index}, options);
+		const getValueOptions = {previousCorrected, index, ...options};
```

Comment:
I replace all Object.assign by spread opperator, spread opperator maintain types, Object.assign break them.

spread opperator can be transpile as Object.assign if needed.

Reply:
spread operator is ok